libdnf/base/goal.cpp

```diff
@@ -45,8 +45,26 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 namespace {
 
 void add_obsoletes_to_data(const libdnf::rpm::PackageQuery & base_query, libdnf::rpm::PackageSet & data) {
+    libdnf::rpm::PackageQuery data_query(base_query.get_base(), libdnf::sack::ExcludeFlags::APPLY_EXCLUDES, true);
```

Comment:
This is not a problem of the PR - The creation of an empty query was cryptic to me.

Reply:
What about to have query constructor with packageset and flags?

Reply:
Yup. The reason is to by-pass the missing constructor.